lib/SampleService/core/validator/builtin.py

```diff
@@ -314,3 +317,51 @@ def _is_num(name, val):
     if val is not None and type(val) != float and type(val) != int:
         raise ValueError(f'Value for {name} parameter is not a number')
     return val
+
+def ontology_has_ancestor(d: Dict[str, Any]) -> Callable[[str, Dict[str, PrimitiveType]], Optional[str]]:
+    '''
+    Build a validation callable that checks that value has valid ontology ancestor term provided
+
+    The 'ontology' parameter is required and must be a string. It is the ontology name.
+
+    The 'ancestor_term' parameter is required and must be a string. It is the ancestor name.
+
+    :param d: the configuration map for the callable.
+    :returns: a callable that validates metadata maps.
+    '''
+    _check_unknown_keys(d, {'ontology', 'ancestor_term'})
+
+    ontology = d.get('ontology')
+    if not ontology:
+        raise ValueError('ontology is a required paramter')
+    if type(ontology) != str:
+        raise ValueError('ontology must be a string')
+
+    ancestor_term = d.get('ancestor_term')
+    if not ancestor_term:
+        raise ValueError('ancestor_term is a required paramter')
+    if type(ancestor_term) != str:
+        raise ValueError('ancestor_term must be a string')
+
+    def _get_ontology_ancestors(ontology, val):
+        srv_wiz_url=os.environ.get('KBASE_ENDPOINT', 'https://ci.kbase.us/services').strip('/') + '/service_wizard'
+        try:
+            oac=OntologyAPI(srv_wiz_url)
+            ret=oac.get_ancestors({"id": val, "ns": ontology})
```

Comment:
How many ancestors is this likely to return? If it's large, it might be worthwhile adding a `has_ancestor` method to the ontology API that simply returns a boolean rather than doing the analysis here.

Reply:
Normally it should get just a few ancestors (<10)